packages/ag-charts-community/src/chart/agChartV2.ts

```diff
@@ -397,16 +398,21 @@ function applyChartOptions(chart: Chart, processedOptions: ProcessedOptions, use
             forceNodeDataRefresh = true;
         }
     }
-    applyLegend(chart, processedOptions);
 
     const seriesOpts = processedOptions.series as any[];
     const seriesDataUpdate = !!processedOptions.data || seriesOpts?.some((s) => s.data != null);
-    const otherRefreshUpdate = processedOptions.legend ?? processedOptions.title ?? processedOptions.subtitle;
-    forceNodeDataRefresh = forceNodeDataRefresh || seriesDataUpdate || !!otherRefreshUpdate;
+    const legendKeys = getLegendKeys();
+    const optionsHaveLegend = Object.values(legendKeys).some(
+        (legendKey) => (processedOptions as any)[legendKey] != null
+    );
```

Comment:
Just to confirm, this condition is only true when `legend` or it's child options get changed, not when `legend` is present but unchanged?

Reply:
`legendKeys` means not the properties of the legend, but `legend`, `gradientLegend` option keys. The previous condition was:
```ts
const otherRefreshUpdate = processedOptions.gradientLegend ?? processedOptions.legend ?? processedOptions.title ?? processedOptions.subtitle;
forceNodeDataRefresh = forceNodeDataRefresh || seriesDataUpdate || !!otherRefreshUpdate;
```